include/fastdds/rtps/attributes/RTPSParticipantAttributes.h

```diff
@@ -301,25 +301,50 @@ class DiscoverySettings
      * Get the static endpoint XML filename
      * @return Static endpoint XML filename
      */
+    FASTRTPS_DEPRECATED("Use static_edp_xml_content()")
     const char* getStaticEndpointXMLFilename() const
     {
-        return m_staticEndpointXMLFilename.c_str();
+        return static_edp_xml_content_.c_str();
     }
 
     /**
      * Set the static endpoint XML filename
      * @param str Static endpoint XML filename
+     * @deprecated
      */
+    FASTRTPS_DEPRECATED("Use static_edp_xml_content()")
     void setStaticEndpointXMLFilename(
             const char* str)
     {
-        m_staticEndpointXMLFilename = std::string(str);
+        static_edp_xml_content_ = "file://" + std::string(str);
+    }
+
+    /**
+     * Set the static endpoint XML content.
+     * @param str URI specifying the static endpoint XML content.
+     * The string could contain a filename (file://) or the XML content directly (data://).
+     */
+    void static_edp_xml_content(
```

Comment:
Knowing that you were not convinced with the name, I propose an alternative. Mind that you'd need to propagate the change
```suggestion
    void static_edp_xml_config(
```